### DISCUSSION TITLE
Support for env.{enviroment-name}

### DISCUSSION THREAD

**Discussion by quornik** [payloadcms/payload#442] [Feature Requests & Ideas]
As in react / next.js it is a nice feature to support the env.envname convention. At this point, Payload does not support for ex. env.local and suggest to gitignore the env at all. Some of the vars we need are safe to be stored in repo, and that is how we are customed to use .env for.

**Comment by jmikrut** [comment]
Hey @quornik — great call! Although, as Payload "plugs in" to your own Node apps, this is actually something you'd need to do yourself completely outside of Payload. 

We actually don't load `.env` at all—instead, that's up to you and typically people will use `dotenv` to do so.

So, to do what you're looking for is pretty easy. Wherever you're using `dotenv` in your app, you could just replace it with something like this:

```js
require('dotenv').config({ path: `.env.${process.env.NODE_ENV}` })
```

Boom! Done! Now, the `.env` file will be loaded dynamically based on the environment that you're running in and you can safely check in an `.env.development` or similar directly into your repo.

Is that what you're looking for?
